components/DrawControls.svelte

```diff
@@ -22,35 +23,51 @@
     map,
   } from "../stores.js";
 
-  const color = "black";
   const circleRadius = 7;
   const lineWidth = 10;
+  const polygonOpacity = 0.5;
+  // TODO The fallback black should never be used in practice, but it seems to
+  // be for polygons being drawn
+  const colorByInterventionType = [
+    "match",
+    ["get", "user_intervention_type"],
+    "area",
+    colors.area,
+    "route",
+    colors.route,
+    "crossing",
+    colors.crossing,
+    "other",
+    colors.other,
+    "black",
+  ];
   const styles = [
     {
-      id: "base-points",
+      id: "interventions-points",
       filter: ["all", isPoint, ["==", "meta", "feature"]],
-      ...drawCircle(color, circleRadius),
+      ...drawCircle(colorByInterventionType, circleRadius),
+      // TODO Outline?
     },
     {
-      id: "draggable-points",
-      filter: ["all", isPoint, ["!=", "meta", "feature"]],
-      // TODO The 1.5 is bulky and ugly, but I can't figure out how to get z-ordering working
-      ...drawCircle("blue", 1.5 * circleRadius),
+      id: "interventions-lines",
+      filter: ["all", isLine, ["==", "meta", "feature"]],
+      ...drawLine(colorByInterventionType, lineWidth),
     },
     {
-      id: "base-line",
-      filter: isLine,
-      ...drawLine(color, lineWidth),
+      id: "interventions-polygons",
+      filter: ["all", isPolygon, ["==", "meta", "feature"]],
+      ...drawPolygon(colorByInterventionType, polygonOpacity),
+      // TODO Outline too?
     },
     {
-      id: "base-polygon-fill",
-      filter: isPolygon,
-      ...drawPolygon(color, 0.1),
+      id: "draggable-points",
+      filter: ["all", isPoint, ["!=", "meta", "feature"]],
```

Comment:
See https://github.com/mapbox/mapbox-gl-draw/blob/main/docs/API.md#styling-draw for some reference. There's a few weird things happening for us right now. If you draw a new polygon, you can see some of them (the fill color being the fallback black, and intermediate points vanishing). I still need to understand all of the combinations of things mapbox-gl-draw can create.